.github/workflows/src/arm-auto-signoff/trivial-changes-check.js

```diff
@@ -0,0 +1,459 @@
+  const git = simpleGit(options.cwd);
+
+  // Filter to only resource-manager files for ARM auto-signoff
+  const changedRmFiles = changedFiles.filter(resourceManager);
+  const changedRmFilesStatuses = {
+    additions: changedFilesStatuses.additions.filter(resourceManager),
+    modifications: changedFilesStatuses.modifications.filter(resourceManager),
+    deletions: changedFilesStatuses.deletions.filter(resourceManager),
+    renames: changedFilesStatuses.renames.filter(
+      (r) => resourceManager(r.to) || resourceManager(r.from),
+    ),
+  };
+
+  // Early exit if no resource-manager changes
+  if (changedRmFiles.length === 0) {
+    core.info("No resource-manager changes detected in PR");
+    core.info(`PR Changes: ${JSON.stringify(changes)}`);
+    core.info(`Is trivial: ${isTrivialPullRequest(changes)}`);
+    return JSON.stringify(changes);
+  }
+
+  // Check if PR contains non-resource-manager changes
+  const hasNonRmChanges = changedFiles.some((file) => !resourceManager(file));
+  if (hasNonRmChanges) {
+    const nonRmFiles = changedFiles.filter((file) => !resourceManager(file));
+    core.info(
+      `PR contains ${nonRmFiles.length} non-resource-manager changes - not eligible for ARM auto-signoff`,
+    );
+    core.info(
+      `Non-RM files: ${nonRmFiles.slice(0, 5).join(", ")}${nonRmFiles.length > 5 ? ` ... and ${nonRmFiles.length - 5} more` : ""}`,
+    );
+    changes.other = true;
+    core.info(`PR Changes: ${JSON.stringify(changes)}`);
+    core.info(`Is trivial: ${isTrivialPullRequest(changes)}`);
+    return JSON.stringify(changes);
+  }
+
+  // Check for non-trivial file operations (additions, deletions, renames)
+  const hasNonTrivialFileOperations = checkForNonTrivialFileOperations(
+    changedRmFilesStatuses,
+    core,
+  );
+  if (hasNonTrivialFileOperations) {
+    core.info("Non-trivial file operations detected (new spec files, deletions, or renames)");
+    // These are functional changes by policy
+    changes.functional = true;
+    core.info(`PR Changes: ${JSON.stringify(changes)}`);
+    core.info(`Is trivial: ${isTrivialPullRequest(changes)}`);
+    return JSON.stringify(changes);
+  }
+
+  // Analyze what types of changes are present and update the changes object
+  await analyzePullRequestChanges(changedRmFiles, git, core, changes);
+
+  core.info(`PR Changes: ${JSON.stringify(changes)}`);
+  core.info(`Is trivial: ${isTrivialPullRequest(changes)}`);
+
+  return JSON.stringify(changes);
+}
+
+/**
+ * Checks for non-trivial file operations: additions/deletions of spec files, or any renames
+ * @param {{additions: string[], modifications: string[], deletions: string[], renames: {from: string, to: string}[]}} changedFilesStatuses - File status information
+ * @param {typeof import("@actions/core")} core - Core logger
+ * @returns {boolean} - True if non-trivial operations detected
+ */
+function checkForNonTrivialFileOperations(changedFilesStatuses, core) {
```

Comment:
```suggestion
function hasSignificantFileOperations(changedFilesStatuses, core) {
```

"has" is a more specific verb than "checkFor", indicating whether "present" returns "true" or "false".  "significant" (or a similar word) is better than "non-trivial", since negative words in names can lead to double-negatives in code, like `hasSignificantFileOperations()` is clearer than `!hasNonTrivialFileOperations()`.